core/templates/dev/head/exploration_editor/ExplorationEditor.js

```diff
@@ -923,17 +923,7 @@ oppia.controller('ExplorationSaveAndPublishButtons', [
 
         modalInstance.result.then(function(commitMessage) {
           _modalIsOpen = false;
-          saveDraftToBackend(commitMessage, function() {
-            explorationData.getLastSavedData().then(function(data) {
-              explorationData.getData().then(function(currentData) {
-                if (data.version > currentData.version) {
-                  autosaveInfoModalsService.showVersionMismatchModal(
-                    changeListService.getChangeList());
-                  return;
-                }
-              });
-            });
-          });
+          saveDraftToBackend(commitMessage);
```

Comment:
Yes that sounds like a good idea. The question of version validity is the same in both cases so it makes sense for the two backend handlers to to treat it in the same way. Alternatively they could both throw an exception; either makes sense provided they both do it!